sklearn/model_selection/_split.py

```diff
@@ -416,6 +420,11 @@ def _iter_test_indices(self, X, y=None, groups=None):
         if self.shuffle:
             check_random_state(self.random_state).shuffle(indices)
 
+        if self.n_splits == 3:
```

Comment:
This is the wrong place to warn. It should happen in check_cv. It should also not warn if the user explicitly sets cv to 3

Reply:
Moved it to check_cv. The tests passed locally.